simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java

```diff
@@ -125,31 +125,53 @@ public void clear() {
   public Enumeration<Collector.MetricFamilySamples> metricFamilySamples() {
     return new MetricFamilySamplesEnumeration();
   }
+
+  public Enumeration<Collector.MetricFamilySamples> metricFamilySamples(Set<String> includedNames) {
+    return new MetricFamilySamplesEnumeration(includedNames);
+  }
+
   class MetricFamilySamplesEnumeration implements Enumeration<Collector.MetricFamilySamples> {
 
     private final Iterator<Collector> collectorIter = collectors().iterator();
     private Iterator<Collector.MetricFamilySamples> metricFamilySamples;
     private Collector.MetricFamilySamples next;
+    private Set<String> includedNames;
 
-    MetricFamilySamplesEnumeration() {
+    MetricFamilySamplesEnumeration(Set<String> includedNames) {
+      this.includedNames = includedNames;
       findNextElement();
     }
 
+    MetricFamilySamplesEnumeration() {
+      this(Collections.<String>emptySet());
+    }
+
     private void findNextElement() {
       if (metricFamilySamples != null && metricFamilySamples.hasNext()) {
-        next = metricFamilySamples.next();
+        next = filter(metricFamilySamples.next());
       } else {
         while (collectorIter.hasNext()) {
           metricFamilySamples = collectorIter.next().collect().iterator();
           if (metricFamilySamples.hasNext()) {
-            next = metricFamilySamples.next();
+            next = filter(metricFamilySamples.next());
+            if(next == null) {
+              continue;
```

Comment:
This will skip the rest of the collector the first time it comes to a non-matching sample.